components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -441,12 +459,16 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
         ideUrl: URL,
         ownerToken: String,
         connectParams: ConnectParams
-    ): String? {
-        var resolveJoinLinkUrl = "https://24000-${ideUrl.host}/joinLink"
+    ): JoinLinkResp? {
+        var resolveJoinLinkUrl = "https://24000-${ideUrl.host}/joinLink2"
```

Comment:
one important thing that it should be backward compatible and work even if joinLink 2 is not available?

Should we have better name, like hostStatus or so?

Reply:
Then better to split this PR in 2? one to just add the endpoint and then wait until it's deployed in all cells and then the other to update and publish a new version of the extension?

Reply:
We cannot rely on cell deployments. It can take a while. We should train ourself to be backward compatible with them. More cells we will have less control will we have.

Reply:
Added bckwards compatible logic